webrtc.html

```diff
@@ -509,33 +509,43 @@
 
         <ol>
           <li>
-            <p>Append an object representing the current call being handled
-            (i.e. function name and corresponding arguments) to the
-            <code>operations</code> array.</p>
+            <p>If <var>connection</var>'s <a href=
+              "#dom-peerconnection-signaling-state"><code>RTCPeerConnection</code>
+              signalingState</a> is <code>closed</code>, throw an
+              <code>InvalidStateError</code> exception and abort these steps.</p>
+          </li>
+
+          <li>
+            <p>Let <var>p</var> be the promise that the called
+            function will return.
+            </p>
+          </li>
+            <p>Add <code>p</code> to <var>operations</var>.</p>
           </li>
 
           <li>
-            <p>If the length of the <code>operations</code> array is exactly 1,
-            execute the function from the front of the queue
-            asynchronously.</p>
+            <p>If the length of <var>operations</var> is exactly 1,
+            execute the steps necessary to resolve <code>p</code>.</p>
           </li>
 
           <li>
-            <p>When the asynchronous operation completes (either successfully
-            or with an error), remove the corresponding object from the
-            <code>operations</code> array. After removal, if the array is
-            non-empty, execute the first object queued asynchronously and
-            repeat this step on completion.</p>
+            <p>When a promise in <code>operations</code> is resolved,
+            remove it from <code>operations</code>.  After removal,
+            if <code>operations</code> is non-empty, execute the steps
+            necessary to resolve the first promise
+            in <code>operations</code>.
```

Comment:
Finally!  You've given me a case when all this chaining stuff you were writing might actually mean something.  Do you think you could create an example where this would matter, perhaps something like this?

console.log(pc.signalingState);
pc.setLocalDescription.then(=> console.log("set local"));
console.log(pc.signalingState);
pc.setRemoteDescription.then(=> console.log("set remote"));

That should print:
stable


This is exactly the problem with your PR 307, and relying 

  Since there's no way I'm going to actually implement PeerConnection's methods in terms of promise chaining, instead I'll have to go read the description and try to figure out what it means and then implement how I think it sounds.  But how it sounds does not convey this incredibly subtle meaning.  Which means the implementation wouldn't actually match the spec, because the spec was written with very subtle behavior expectations that would only make sense if you wrote it that on specific way.  In other words, by describing exactly how to do something rather than what to do, you made it so specific that unless the implementor followed your exact implementation, they would never